internal/controllers/build_sign_events_reconciler.go

```diff
@@ -211,3 +205,41 @@ func (r *JobEventReconciler) removeFinalizer(ctx context.Context, pod *v1.Pod) e
 
 	return nil
 }
+
+//go:generate mockgen -source=build_sign_events_reconciler.go -package=controllers -destination=mock_build_sign_events_reconciler.go JobEventReconcilerHelper
+
+type JobEventReconcilerHelper interface {
+	GetOwner(context.Context, metav1.OwnerReference, string) (client.Object, error)
+}
+
+type jobEventReconcilerHelper struct {
+	client client.Client
+}
+
+func NewJobEventReconcilerHelper(client client.Client) JobEventReconcilerHelper {
+	return &jobEventReconcilerHelper{client: client}
+}
+
+func (h *jobEventReconcilerHelper) GetOwner(ctx context.Context, ref metav1.OwnerReference, namespace string) (client.Object, error) {
+	owner := &unstructured.Unstructured{}
+	owner.SetKind(ref.Kind)
```

Comment:
WDYT about checking the Kind (module, hub, preflight)? If it is not one of them, then we can log an error. In addition, you can skip the checking the namespace of the owner: hub and preflight a non-namspaced and module is the same namespace as pod

Reply:
> WDYT about checking the Kind (module, hub, preflight)? If it is not one of them, then we can log an error.

I think it's more future-proof if we do not even look at the owner's kind.

> In addition, you can skip the checking the namespace of the owner: hub and preflight a non-namspaced and module is the same namespace as pod

We need to handle both namespace-scoped (`Module`) and cluster-scoped (`PreflightValidation`, `ManagedClusterModule`) owners.

Reply:
if we check the kind, then we know if the object is namespaced or not (preflight, manager-hub), and if it is namespaced (module), then the namespace should be the same as the build/sign pod, no need to search for it. But you foresee additional objects being used as onwer, then your code is better

Reply:
> if we check the kind, then we know if the object is namespaced or not

`client.IsObjectNamespaced` does that for us, and I'd rather use it than rely on our own `switch`. I don't think that looking for our own kinds would make the code significantly more straightforward, and we would have to change it if we added any new owning object.